src/components/MultiCombobox/MultiCombobox.test.tsx

```diff
@@ -224,4 +229,32 @@ describe('MultiCombobox', () => {
     });
     expect(queryAllByRole('tag')).toHaveLength(2);
   });
+
+  it('works with grouped items', async () => {
+    const { getByText, getByPlaceholderText } = renderWithTheme(
```

Comment:
you  are not actually checking that the grouping worked here. I think you should add some `aria-*` attributes in the "nested/grouped" items to signify the group they belong in.

By doing this, you'll then be able to check for these attributes in your tests and verify that indeed grouping worked